### PR TITLE
feat: Add ignoreCommits config value to skip single commit linting when unnecessary

### LINKED ISSUE #17: Setting commitTitleMatch to false does not disable commit message linting
## Expected Behavior
Setting `commitTitleMatch: 'false'` should stop the linter from checking commits within a PR.

## Actual Behavior
For our use case, because we have Github set to use the PR title for the squash merges, we have no need to ensure that the first (or any) commit follows the linting spec, however with `commitTitleMatch: 'false'` set, we still receive errors for PRs with a single commit when some part of that single commit does not follow the linting rules.

```
Error: COMMIT: PRs with a single commit require the commit message to conform to the conventional commit spec
```

In the case of the above error, it was a Dependabot PR with a single commit which actually passed correctly for the title, but was failing on;
> Error: Commit message: body's lines must not be longer than 100 characters

## Steps to Reproduce the Problem

```
name: Lint PR Title

on:
  pull_request:
    branches:
      - development
    types:
      - opened
      - edited

jobs:
  pr-lint:
    runs-on: ubuntu-latest
    steps:
      - name: Check Out
        uses: actions/checkout@v3

      - name: Lint PR Title
        uses: CondeNast/conventional-pull-request-action@v0.1.2
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          commitlintRulesPath: './commitlint.rules.js'
          commitTitleMatch: 'false'
```

I think the issue here, as outlined in #18 is that even with the `commitTitleMatch` set to `false`, the linter will still lint the content of the first single commit to the default commitlint ruleset, and only skips the check that ensures it matches the title. This is desirable for users who merge commit, but don't care about their PR title, but for reverse situations like the above it would be preferable to have an option to completely skip all commit message linting when their content is irrelevant.

### PR DESCRIPTION
**Pull request by benhodgson87**
Fixes #17 

For repos where the strategy is to squash merge all commits to a branch, and Github is set up to use the PR title as the merge commit title, this action currently always lints single commits, even if their content is irrelevant.

This PR adds a new config value, `ignoreCommits`, which forces the tool to always lint the title of the PR, and ignore the validity of any commits that may be made.

This allows users who merge commit to still use `commitTitleMatch` for match checking, but users who squash with the title to completely disable these checks.

## Review Checklist

- [x] This PR provides a description, testing steps, and adequate context for reviewers
- [x] Unit tests added/updated to provide coverage for functional code changes in this PR

### CODE REVIEW

**Comment by mattbedell** [comment]
I'm hesitant to add a setting that allows the pre-populated merge commit message to be incorrect. In the linked issue, would changing this [line](https://github.com/CondeNast/conventional-pull-request-action/blob/main/src/lint-pr.js#L54) to lint only the [`commitMessageSubject`](https://github.com/CondeNast/conventional-pull-request-action/blob/main/src/lint-pr.js#L52), fix the bug you encountered?

File: README.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ jobs:
           commitlintRulesPath: "./commitlint.rules.js" # default: undefined
           # if the PR contains a single commit, fail if the commit message and the PR title do not match
           commitTitleMatch: "true" # default: 'true'
+          # if you squash merge PRs and have your github repo set to always use the PR title as the merge commit, you can disable all linting of commits
```
Comment (crccheck) [inline README.md:60]: It'll help if this uses the exact same language as the Github UI (at least in English):
```suggestion
          # if you squash merge PRs and enabled "Default to PR title for squash merge commits", you can disable all linting of commits
```

**Comment by benhodgson87** [comment]
> I'm hesitant to add a setting that allows the pre-populated merge commit message to be incorrect. In the linked issue, would changing this [line](https://github.com/CondeNast/conventional-pull-request-action/blob/main/src/lint-pr.js#L54) to lint only the [`commitMessageSubject`](https://github.com/CondeNast/conventional-pull-request-action/blob/main/src/lint-pr.js#L52), fix the bug you encountered?

The issue here is this still requires the first commit message to conform to (for us) arbitrary linting requirements (eg. requiring users to mark their first commit with the type). For our use case - we just need the squashed PR title to include the type for semantic release versioning.

As an example of our problem with this, in some cases we have situations where engineers (and often non-engineers making small copy changes to markdown through the Github web UI) will forget to add the type to their first commit, as we traditionally mark our PR types through branch naming and other projects internally don't yet follow conventional commits, which then requires them to work out what's gone wrong, amend the commit message, and push it up just to pass linting, even if the PR title is actually perfectly correct. In cases where this happens with copy changes through the web UI it usually requires the approving codeowner to pull the branch, amend the commit, and push it up to pass, only for that commit to be discarded at merge.

I think a well documented escape hatch here does make sense, but that said we've been using the fork within our repo to get this functionality and it's been working well enough, I realise this may be an 'us' problem so I'm happy to continue running from that and close this if the use case doesn't fit the scope here.

**Comment by mattbedell** [comment]
@benhodgson87 makes sense, apologies for the long turn around time

**Pull request merged**